cave/plot/algorithm_footprint.py

```diff
@@ -508,7 +508,7 @@ def get_clusters(self, features_2d):
         # get silhouette scores for k_means with 2 to 12 clusters
         # use number of clusters with highest silhouette score
         best_score, best_n_clusters = -1, -1
-        min_clusters, max_clusters = 2, 12
+        min_clusters, max_clusters = 2, min(features_2d.shape[0], 12)
```

Comment:
Why is this change necessary?

Reply:
If there are less than 12 instances, this throws an Error.

Reply:
Ok then it makes total sense